spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -536,7 +539,7 @@ protected <T> void commit(SnapshotUpdate<T> snapshotUpdate, long epochId, String
     }
 
     private Long findLastCommittedEpochId() {
-      Snapshot snapshot = table.currentSnapshot();
+      Snapshot snapshot = table.snapshot(branch);
```

Comment:
Shouldn't this be current? Branch could be null.

Reply:
using `SparkUtil.latestSnapshot(table, branch)` for this